src/common/cockpitwebserver.c

```diff
@@ -837,8 +822,13 @@ cockpit_web_request_process (CockpitWebRequest *self,
       return;
     }
 
-  self->original_path = path;
-  self->path = path + self->web_server->url_root->len;
+  self->original_path = g_strdup (path);
+  self->path = self->original_path + self->web_server->url_root->len;
+  self->query = strchr (self->path, '?');
+  if (self->query)
+    *((gchar *) self->query++) = '\0';
```

Comment:
As this gets modified, why not drop the `const` from the declaration in cockpitwebrequest-private.h and get rid of the cast here? But this is still a bit confusing, as this is now really just a pointer into the `path`, which is also `const gchar*`. But then, `path` by itself just points into `original_path`, which at last is writable. 

This is all a bit hard to follow, TBH. Would it be possible to flip this around, so that `original_path` is const, and actually never modified? and `path` gets strdup()'ed, truncated etc.? `query` should stay const I think, due to the `else` branch.

Reply:
This was another case of "...I rewrote this so many freaking times..."

That's a good suggestion, though.  I'll look into it.

Reply:
I remember why I did this now: `path` gets assigned to from string literals in very many testcases.  If we change it to `gchar *`, then we get errors all over the place about dropping the `const` qualifier.

Will keep thinking about this...